src/components/StripeTerminalProvider.tsx

```diff
@@ -203,6 +208,30 @@ export function StripeTerminalProvider({
     [log]
   );
 
+  const didChangeOfflineStatus = useCallback(
+    ({ result }: { result?: OfflineStatus }) => {
+      log('didChangeOfflineStatus');
+      emitter?.emit(CHANGE_OFFLINE_STATUS, result);
+    },
+    [log]
+  );
+
+  const didForwardPaymentIntent = useCallback(
+    ({ result }: EventResult<PaymentIntent.Type>) => {
```

Comment:
We probably want this `EventResult` to be an object with a `payment_intent` and an `error` key. We do want to support returning both.